main.go

```diff
@@ -89,12 +90,50 @@ func main() {
 		Namespace:              watchNamespace,
 	})
 	if err != nil {
-		setupLog.Error(err, "unable to start manager")
+		setupLog.Error(err, "unable to create manager")
 		os.Exit(1)
 	}
 
+	ctx := ctrl.SetupSignalHandler()
+
+	// Fetch ClientConfigs and create the clientCache
 	clientCache := clientcache.New(mgr.GetClient(), scheme)
 
+	cConfigs := k8ssandraiov1alpha1.ClientConfigList{}
+	err = mgr.GetClient().List(ctx, &cConfigs)
```

Comment:
It would be good to put this code and the following for loop into some functions/methods, maybe in clientcache so that we can write some integration tests for it. I am fine doing it in a follow up PR though if need be rather. I want to get this merged sooner rather than later.